taichi/transforms/compile_to_offloads.cpp

```diff
@@ -171,6 +171,7 @@ void offload_to_executable(IRNode *ir,
   irpass::demote_operations(ir);
   print("Operations demoted");
 
+  irpass::type_check(ir);
```

Comment:
This should be moved into `demote_operations`. It's better to have each IR pass emitting type-checked and verified IR.